homeassistant/components/bryant_evolution/climate.py

```diff
@@ -96,106 +105,33 @@ def __init__(
             via_device=(DOMAIN, names.system_device_uid(sam_uid, system_id)),
             name=f"System {system_id} Zone {zone_id}",
         )
+        self._set_attrs_from_coordinator()
 
-    async def async_update(self) -> None:
-        """Update the entity state."""
-        self._attr_current_temperature = await self._client.read_current_temperature()
-        if (fan_mode := await self._client.read_fan_mode()) is not None:
-            self._attr_fan_mode = fan_mode.lower()
-        else:
-            self._attr_fan_mode = None
-        self._attr_target_temperature = None
-        self._attr_target_temperature_high = None
-        self._attr_target_temperature_low = None
-        self._attr_hvac_mode = await self._read_hvac_mode()
-
-        # Set target_temperature or target_temperature_{high, low} based on mode.
-        match self._attr_hvac_mode:
-            case HVACMode.HEAT:
-                self._attr_target_temperature = (
-                    await self._client.read_heating_setpoint()
-                )
-            case HVACMode.COOL:
-                self._attr_target_temperature = (
-                    await self._client.read_cooling_setpoint()
-                )
-            case HVACMode.HEAT_COOL:
-                self._attr_target_temperature_high = (
-                    await self._client.read_cooling_setpoint()
-                )
-                self._attr_target_temperature_low = (
-                    await self._client.read_heating_setpoint()
-                )
-            case HVACMode.OFF:
-                pass
-            case _:
-                _LOGGER.error("Unknown HVAC mode %s", self._attr_hvac_mode)
-
-        # Note: depends on current temperature and target temperature low read
-        # above.
-        self._attr_hvac_action = await self._read_hvac_action()
-
-    async def _read_hvac_mode(self) -> HVACMode:
-        mode_and_active = await self._client.read_hvac_mode()
-        if not mode_and_active:
-            raise HomeAssistantError(
-                translation_domain=DOMAIN, translation_key="failed_to_read_hvac_mode"
-            )
-        mode = mode_and_active[0]
-        mode_enum = {
-            "HEAT": HVACMode.HEAT,
-            "COOL": HVACMode.COOL,
-            "AUTO": HVACMode.HEAT_COOL,
-            "OFF": HVACMode.OFF,
-        }.get(mode.upper())
-        if mode_enum is None:
-            raise HomeAssistantError(
-                translation_domain=DOMAIN,
-                translation_key="failed_to_parse_hvac_mode",
-                translation_placeholders={"mode": mode},
-            )
-        return mode_enum
+    def _set_attrs_from_coordinator(self) -> None:
+        # Propagate some parameters that are really system-level, not zone-level,
+        # but that the climate entity needs.
+        self._attr_fan_mode = self.coordinator.read_fan_mode(self.system_id)
```

Comment:
Is the idea that there's no need for a _handle_coordinator_update function? You'd just have an @property for every property, and somehow the data update coordinator ensures that new entity state is written to HA? That seems different than the docs and other code I saw if so, but I'm open to discussion.

Reply:
Yes, the coordinator will inform all listeners that it has new values and they will then retrieve the data from the coordinator and the new states are written to the state machine

Reply:
OK, I think I understand what you are suggesting, but I don't think it will work for this integration.

One of the (irritating!) things about the device this integration controls is that reads and writes to it are very slow -- about 1500/ms per parameter (hvac mode, temperature, etc). Because of that, whenever we make a write to the device, we also update our own state locally. E.g., when we set the fan mode:

```
async def async_set_fan_mode(self, fan_mode: str) -> None:
        """Set new target fan mode."""
        client = await self._client()
        if not await client.set_fan_mode(fan_mode):
            raise HomeAssistantError(
                translation_domain=DOMAIN, translation_key="failed_to_set_fan_mode"
            )
        self._attr_fan_mode = fan_mode.lower(). # ****Local update here***
        self.async_write_ha_state()
        await self.coordinator.async_request_refresh()
```

The reason we do this is to avoid the controls flicking back to the last-read state from the coordinator until the next update finishes.

If we have the climate entity just read values directly from the coordinator, then this wouldn't work.

Reply:
Nope, you still don't understand. What i'm tryng to say is that it will **NOT** flick back 🙄
Try it, you will see. I did, added even a sleep of 10 seconds, it did not flick back.

Reply:
I just modified the code running against the actual device to use a property function for the fan mode and to not set fan mode in _set_attrs_from_coordinator.

With the fan mode in auto, I changed it to low using the entity card. I then reloaded the page.

On reload, the entity card displayed fan mode auto. The actual thermostat (the hardware that's on the wall) showed fan mode low. The next time the HA integration polled (through the coordinator), the entity card switch to showing fan mode low.

This seems consistent with my understanding of how the code is working -- the entity state doesn't actually change until the next poll, so if we want to have the changes that we know we execute become visible immediatel,y the code needs to be structured as-is.

Happy to chat on discord it'd make it easier to come to a mutual understanding here.

Reply:
New suggestion 

```suggestion
        self._attr_fan_mode = self.system.fan_mode
```

Define `self.system = coordinator.data.systems[sys_id]` and maybe also `self.zone = coordinator.data.systems[sys_id].zones[zone_id]` in the init for better readability